torch/csrc/api/include/torch/optim/schedulers/lr_scheduler.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/optim/optimizer.h>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+namespace torch {
+namespace optim {
+
+class TORCH_API LRScheduler {
+public:
+
+  //This class needs to take a reference of an optimizer from outside such that it can
+  //modify its learning rates; due to this the lifetime of said optimizer must be maintained
+  LRScheduler(torch::optim::Optimizer& optimizer);
+
+  void step();
+
+protected:
+  //A vector of learning rates is calculated and returned from the specific subclass.
+  //A vector is returned with each element being a separate learning rate for each
+  //param group - although the normal use case would be to return a vector of identical
+  //elements.
+  virtual std::vector<double> get_lrs() = 0;
```

Comment:
Since you have this virtual function, you need to declare the destructor of this class as virtual, and the derived class (eg StepLR) will has virtual destructor automatically. Please change.